subsys/cpp/Kconfig

```diff
@@ -54,6 +54,7 @@ if LIB_CPLUSPLUS
 
 config EXCEPTIONS
 	bool "Enable C++ exceptions support"
+	depends on ARC || ARM || (RISCV && !64BIT) || (X86 && !64BIT) || XTENSA
```

Comment:
I was thinking the same in #35647, but are we sure that the non-working platforms is a Zephyr issue and not a toolchain (newlib) issue ?

If this could be an issue with the `Zephyr-sdk` and its newlib, then it would in principle be wrong to exclude `x86_64`, `riscv64`, and `qemu_cortex_a53` if they have the potential to work if compiled with another toolchain.

Reply:
At least, for ARM64 (`qemu_cortex_a53`), I also tested with the GNU Toolchain for AArch64 Baremetal (gcc-arm-10.2-2020.11-x86_64-aarch64-none-elf) from Arm themselves, and C++ exception handling is failing in the identical manner.

Reply:
I'll look into this.  I think its probably the same issue on all 64-bit systems.  I've got a simpler test that seems to pass.

Reply:
So the issue is linking on 64-bit systems on what is happening with eh_frame

Reply:
Wrong `__EH_FRAME_BEGIN__` seems to be the reason apparently:
```
/opt/zephyr-sdk-0.12.4/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-objdump -x zephyr/zephyr.elf | grep eh_frame
 13 .eh_frame     000015f0  000000008001a080  000000008001a080  00016df8  2**3
000000008001a080 l    d  .eh_frame      0000000000000000 .eh_frame
000000008001b66c l     O .eh_frame      0000000000000000 __EH_FRAME_BEGIN__
000000008001b66c l     O .eh_frame      0000000000000000 __FRAME_END__
```

Reply:
Though, it's something else for SPARC -- the `__EH_FRAME_BEGIN__` is correct:
```
 13 .eh_frame     00000c38  40028f48  40028f48  000257f8  2**2
40028f48 l    d  .eh_frame      00000000 .eh_frame
40028f48 l     O .eh_frame      00000000 __EH_FRAME_BEGIN__
40029b7c l     O .eh_frame      00000000 __FRAME_END__
```

Reply:
Ok, so adding `SORT_NONE` to the linker script should fix the problem.